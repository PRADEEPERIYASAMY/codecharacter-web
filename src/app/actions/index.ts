export * from 'app/actions/dashboard';
