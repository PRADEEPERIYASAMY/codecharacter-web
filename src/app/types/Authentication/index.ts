export enum AuthType {
  LOGIN = 'LOGIN',
  REGISTER = 'REGISTER',
}

export interface State {
  isAuthModalOpen: boolean;
  authType: AuthType;
}

export interface StateProps {
  isLoggedIn: boolean;
}

export type Props = StateProps;
