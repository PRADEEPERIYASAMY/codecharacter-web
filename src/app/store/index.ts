import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

export function configureStore(initialState?: object) {
  let middleware = applyMiddleware(logger);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const store = createStore(rootReducer, initialState, middleware);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
