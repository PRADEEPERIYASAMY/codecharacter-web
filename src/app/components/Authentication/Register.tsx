import { faEnvelope, faFlag, faLock, faUser, faUserTie } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { RECAPTCHA_SITE_KEY } from 'app/../config/config';
import * as styles from 'app/styles/Authentication.module.css';
import 'app/styles/Register.css';
import { AuthType } from 'app/types/Authentication';
import * as RegisterInterfaces from 'app/types/Authentication/Register';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';
// tslint:disable-next-line:import-name
import ReactFlagsSelect from 'react-flags-select';
import 'react-flags-select/css/react-flags-select.css';
// tslint:disable-next-line:import-name
import ReCAPTCHA from 'react-google-recaptcha';

export class Register extends React.Component<RegisterInterfaces.Props, RegisterInterfaces.State> {
  private registerRef = React.createRef<HTMLFormElement>();
  private recaptchaRef = React.createRef<ReCAPTCHA>();

  constructor(props: RegisterInterfaces.Props) {
    super(props);

    this.state = {
      country: 'IN',
      email: '',
      fullName: '',
      isCaptchaValidated: false,
      password: '',
      pragyanId: '',
      repeatPassword: '',
      username: '',
    };
  }

  public render() {
    const { repeatPassword, email, password, username, fullName } = this.state;
    const { handleSelectPanel, checkUsernameExists, errorMessage, updateErrorMessage } = this.props;
    return (
      <div>
        <Row>
          <div className={classnames('col-sm-10 offset-sm-1', styles.form)}>
            <form
              className={'registerForm'}
              noValidate
              ref={this.registerRef}
              onSubmit={this.handleRegister}
            >
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faUser} />
                      </span>
                    </div>
                    <input
                      type="text"
                      className="form-control"
                      id="registerValidationUsername"
                      placeholder="Username"
                      aria-describedby="inputGroupPrepend"
                      maxLength={50}
                      minLength={5}
                      value={username}
                      onChange={(e) => {
                        checkUsernameExists(e.target.value);
                        this.setState({
                          username: e.target.value,
                        });
                      }}
                      required
                    />
                    <div className="invalid-feedback">Username must have minimum 5 characters.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faUserTie} />
                      </span>
                    </div>
                    <input
                      type="text"
                      className="form-control"
                      id="registerValidationFullname"
                      placeholder="Name"
                      aria-describedby="inputGroupPrepend"
                      maxLength={50}
                      minLength={5}
                      value={fullName}
                      onChange={(e) =>
                        this.setState({
                          fullName: e.target.value,
                        })
                      }
                      required
                    />
                    <div className="invalid-feedback">Name must have minimum 5 characters.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faEnvelope} />
                      </span>
                    </div>
                    <input
                      type="email"
                      className="form-control"
                      id="registerValidationEmail"
                      placeholder="Email"
                      aria-describedby="inputGroupPrepend"
                      value={email}
                      onChange={(e) =>
                        this.setState({
                          email: e.target.value,
                        })
                      }
                      required
                    />
                    <div className="invalid-feedback">Please enter a valid Email ID.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col sm={12} mb-3">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faFlag} />
                      </span>
                    </div>
                    <ReactFlagsSelect
                      searchable={true}
                      placeholder="Search for a country"
                      className="customFlag"
                      defaultCountry="IN"
                      onSelect={this.onSelectFlag}
                    />
                    <div className="invalid-feedback">Please select a country.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="registerValidationPassword"
                      placeholder="Password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      value={password}
                      onChange={(e) =>
                        this.setState({
                          password: e.target.value,
                        })
                      }
                      required
                    />
                    <div className="invalid-feedback">
                      Password should have minimum 5 characters.
                    </div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-1">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="registerValidationrepeatPassword"
                      placeholder="Confirm Password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      value={repeatPassword}
                      onChange={(e) =>
                        this.setState({
                          repeatPassword: e.target.value,
                        })
                      }
                      required
                    />
                    <div className="invalid-feedback">Passwords should match.</div>
                  </div>
                </div>
              </div>
              <div className="form-row d-flex justify-content-center my-1">
                <div className="d-flex justify-content-center input-group">
                  <ReCAPTCHA
                    sitekey={RECAPTCHA_SITE_KEY}
                    data-theme={'dark'}
                    onChange={this.onChange}
                    ref={this.recaptchaRef}
                  />
                  <div className="invalid-feedback">Please fill recaptcha.</div>
                </div>
              </div>
              <div className="form-row">
                <div className="input-group" />
                <div className="col text-center mt -0 mb-2 errorMessage">{errorMessage}</div>
              </div>
              <div className="form-row">
                <div className="col text-center">
                  <button
                    className="btn btn-info"
                    type="submit"
                    style={{
                      width: '100%',
                    }}
                  >
                    Register
                  </button>
                </div>
              </div>
            </form>
          </div>
        </Row>
        <Row
          style={{
            borderTop: '2px solid #999',
          }}
        >
          <Col className="ml-auto  my-3 mr-auto">
            <div className="text-dark">
              Already have an account?{' '}
              <a
                className="text-primary"
                style={{
                  cursor: 'pointer',
                }}
                onClick={() => {
                  updateErrorMessage('');
                  handleSelectPanel(AuthType.LOGIN);
                }}
              >
                Login now
              </a>
            </div>
          </Col>
        </Row>
      </div>
    );
  }

  private onSelectFlag = (countryCode: string) => {
    this.setState({
      country: countryCode,
    });
  };

  private handleRegister = (event: React.FormEvent<HTMLFormElement>) => {
    const { register, handleSelectPanel } = this.props;
    const {
      repeatPassword,
      country,
      email,
      fullName,
      password,
      username,
      pragyanId,
      isCaptchaValidated,
    } = this.state;
    const form = this.registerRef.current;
    event.preventDefault();

    if (form) {
      if (form.checkValidity() && isCaptchaValidated) {
        register({
          country,
          email,
          fullName,
          password,
          pragyanId,
          repeatPassword,
          username,
        });
        handleSelectPanel(AuthType.LOGIN);
      }
      form.classList.add('was-validated');
    }
  };

  private onChange = (value: string | null) => {
    if (value) {
      this.setState({
        isCaptchaValidated: true,
      });
    }
  };
}
