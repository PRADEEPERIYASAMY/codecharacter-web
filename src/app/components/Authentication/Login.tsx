import { faEnvelope, faLock } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { AuthType } from 'app/components/Authentication/';
import * as styles from 'app/styles/Authentication.module.css';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';

export class Login extends React.Component<Login.Props, Login.State> {
  private loginRef = React.createRef<HTMLFormElement>();

  constructor(props: Login.Props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  public render() {
    return (
      <div>
        <Row>
          <div className={classnames('col-sm-10 offset-sm-1', styles.form)}>
            <form
              className={classnames('loginForm')}
              noValidate
              ref={this.loginRef}
              onSubmit={(e) => {
                const form = this.loginRef.current;
                if (form) {
                  if (!form.checkValidity()) {
                    e.preventDefault();
                  }
                  form.classList.add('was-validated');
                }
              }}
            >
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faEnvelope} />
                      </span>
                    </div>
                    <input
                      type="email"
                      className="form-control"
                      id="validationEmail"
                      placeholder="Your email"
                      aria-describedby="inputGroupPrepend"
                      required
                      value={this.state.email}
                      onChange={(e) =>
                        this.setState({
                          email: e.target.value,
                        })
                      }
                    />
                    <div className="invalid-feedback">Please enter a valid Email ID.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="validationPassword"
                      placeholder="Your password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      required
                    />
                    <div className="invalid-feedback">Please enter the correct password.</div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col text-center">
                  <button
                    className="btn btn-info"
                    type="submit"
                    style={{
                      width: '100%',
                    }}
                  >
                    Login
                  </button>
                </div>
              </div>
            </form>
          </div>
        </Row>
        <Row
          style={{
            borderTop: '2px solid #999',
          }}
        >
          <Col className="text-center my-3 ml-auto mr-auto">
            <div className="text-dark">
              Don't have an account?{' '}
              <a
                className="text-primary"
                style={{
                  cursor: 'pointer',
                }}
                onClick={() => this.props.handleSelectPanel(AuthType.REGISTER)}
              >
                Create one
              </a>
            </div>
          </Col>
        </Row>
      </div>
    );
  }
}

export namespace Login {
  export interface OwnProps {
    handleSelectPanel: (authType: AuthType) => void;
  }
  export interface State {
    email: string;
    password: string;
  }
  export type Props = OwnProps;
}
