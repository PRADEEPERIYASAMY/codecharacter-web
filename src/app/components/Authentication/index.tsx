import { faTimes, faUser, faUserPlus } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Login from 'app/containers/Authentication/Login';
import Register from 'app/containers/Authentication/Register';
import * as styles from 'app/styles/Authentication.module.css';
import * as AuthenticationInterfaces from 'app/types/Authentication';
import classnames from 'classnames';
import * as React from 'react';
import { Button, Col, Grid, Row } from 'react-bootstrap';
export class Authentication extends React.Component<
  AuthenticationInterfaces.Props,
  AuthenticationInterfaces.State
> {
  constructor(props: AuthenticationInterfaces.Props) {
    super(props);
    const { isLoggedIn } = this.props;
    this.state = {
      authType: AuthenticationInterfaces.AuthType.REGISTER,
      isAuthModalOpen: !isLoggedIn,
    };
  }

  public render() {
    const { isAuthModalOpen, authType } = this.state;
    return (
      <div
        className={classnames(styles.Authentication, {
          [`${styles.modalOpen}`]: isAuthModalOpen,
          [`${styles.modalClose}`]: !isAuthModalOpen,
        })}
      >
        <Grid
          fluid={true}
          className={classnames(styles.modal, {
            [`${styles.slideEnter}`]: isAuthModalOpen,
            [`${styles.slideExit}`]: !isAuthModalOpen,
          })}
        >
          <Row className="justify-content-between">
            <Col className={'ml-auto'}>
              <Button
                className={classnames(styles.customBtn)}
                bsSize="xsmall"
                onClick={() => this.toggleAuthenticationModal(!isAuthModalOpen)}
              >
                <FontAwesomeIcon icon={faTimes} color={'black'} />
              </Button>
            </Col>
          </Row>
          <Row>
            <Col sm={12}>
              <div className={classnames(styles.buttonPanelBackground)}>
                <Button
                  className={classnames(styles.buttonPanel, {
                    [`${styles.buttonPanelActive}`]:
                      authType === AuthenticationInterfaces.AuthType.LOGIN,
                  })}
                  onClick={() => this.handleSelectPanel(AuthenticationInterfaces.AuthType.LOGIN)}
                >
                  <FontAwesomeIcon icon={faUser} /> &nbsp;Login
                </Button>
                <Button
                  className={classnames(styles.buttonPanel, {
                    [`${styles.buttonPanelActive}`]:
                      authType === AuthenticationInterfaces.AuthType.REGISTER,
                  })}
                  onClick={() => this.handleSelectPanel(AuthenticationInterfaces.AuthType.REGISTER)}
                >
                  <FontAwesomeIcon icon={faUserPlus} /> &nbsp;Register
                </Button>
              </div>
            </Col>
          </Row>
          {authType === AuthenticationInterfaces.AuthType.LOGIN ? (
            <Login handleSelectPanel={this.handleSelectPanel} />
          ) : (
            <Register handleSelectPanel={this.handleSelectPanel} />
          )}
        </Grid>
      </div>
    );
  }

  private toggleAuthenticationModal = (isAuthModalOpen: boolean) => {
    this.setState({
      isAuthModalOpen,
    });
  };
  private handleSelectPanel = (authType: AuthenticationInterfaces.AuthType) => {
    this.setState({
      authType,
    });
  };
}
