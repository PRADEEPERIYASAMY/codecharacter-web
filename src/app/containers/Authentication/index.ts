import { Authentication } from 'app/components/Authentication';
import { RootState } from 'app/reducers';
import * as AuthenticationInterfaces from 'app/types/Authentication/';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    isLoggedIn: rootState.user.isLoggedIn,
  };
};

const authenticationContainer = connect<AuthenticationInterfaces.Props, {}, {}>(mapStateToProps)(
  Authentication,
);

export default authenticationContainer;
