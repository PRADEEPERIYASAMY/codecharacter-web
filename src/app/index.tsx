import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
