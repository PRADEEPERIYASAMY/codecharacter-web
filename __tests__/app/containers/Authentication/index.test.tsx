import Authentication from 'app/containers/Authentication';
import { configureStore } from 'app/store';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('Authentication main Container', () => {
  const { store } = configureStore();
  const wrapper = shallow(<Authentication />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
