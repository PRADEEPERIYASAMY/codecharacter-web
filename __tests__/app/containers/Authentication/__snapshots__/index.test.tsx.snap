// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication main Container Should render 1`] = `
<Authentication
  isLoggedIn={true}
  toggleUserProfileModal={[Function]}
  updateErrorMessage={[Function]}
/>
`;
